pkg/asset/installconfig/aws/subnet.go

```diff
@@ -99,38 +145,62 @@ func subnets(ctx context.Context, session *session.Session, region string, ids [
 		},
 	)
 	if err != nil {
-		return vpc, nil, nil, errors.Wrap(err, "describing route tables")
+		return subnetGroups, errors.Wrap(err, "describing route tables")
+	}
+
+	azs, err := client.DescribeAvailabilityZonesWithContext(
+		ctx,
+		&ec2.DescribeAvailabilityZonesInput{
+			ZoneNames: zoneNames,
+		},
+	)
+	if err != nil {
+		return subnetGroups, errors.Wrap(err, "describing availability zones")
+	}
+	for _, az := range azs.AvailabilityZones {
+		availabilityZones[*az.ZoneName] = az
 	}
 
 	publicOnlySubnets := os.Getenv("OPENSHIFT_INSTALL_AWS_PUBLIC_ONLY") != ""
 
 	for _, id := range ids {
 		meta, ok := metas[id]
 		if !ok {
-			return vpc, nil, nil, errors.Errorf("failed to find %s", id)
+			return subnetGroups, errors.Errorf("failed to find %s", id)
 		}
 
 		isPublic, err := isSubnetPublic(routeTables, id)
 		if err != nil {
-			return vpc, nil, nil, err
+			return subnetGroups, err
 		}
-		if isPublic {
-			public[id] = meta
-		} else {
-			private[id] = meta
+		meta.Public = isPublic
+		meta.ZoneType = *availabilityZones[meta.Zone].ZoneType
+		meta.ZoneGroup = *availabilityZones[meta.Zone].GroupName
+
+		if meta.ZoneType == awstypes.LocalZoneType {
```

Comment:
nit: Instead of `continue` in several places, could this be altered to use if/if-else/else ?

Reply:
@barbacbd that's a good idea. Changed.